spark/v3.0/spark/src/main/java/org/apache/spark/sql/connector/iceberg/read/SupportsFileFilter.java

```diff
@@ -32,5 +32,22 @@
    *
    * @param locations file locations
    */
-  void filterFiles(Set<String> locations);
+  SupportsFileFilter.FileFilterMetric filterFiles(Set<String> locations);
+
+  class FileFilterMetric {
+    private int totalFilesNumber;
+    private int hitFilesNumber;
+
+    public FileFilterMetric(int totalFilesNumber, int hitFilesNumber) {
+      this.totalFilesNumber = totalFilesNumber;
+      this.hitFilesNumber = hitFilesNumber;
+    }
+
+    public int getTotalFilesNumber() {
+      return totalFilesNumber;
+    }
+    public int getHitFilesNumber() {
```

Comment:
Style: there should be a newline separating methods.

Reply:
done